rasheed - rendered the store list

desokan opened the pull request.

desokan committed:
rendered the store list

desokan committed:
added the add to cart function

desokan committed:
added the add to cart function

desokan committed:
rendered the cart list

desokan committed:
completed core

desokan committed:
added comments

desokan committed:
added new comments

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+    // Create a new image element and set the src and alt attributes.
+    const image = document.createElement("img");
+    image.setAttribute("src", `assets/icons/${item.id}.svg`);
+    image.setAttribute("alt", `${item.name}`);
```
Very nice, never thought about adding this!

ChrisJS90 commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+    0
+  );
+
+  const formatCurrency = new Intl.NumberFormat("en-GB", {
```
Did not know about this function but will definitely be using it in the future!

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+  cartUL.innerHTML = "";
+
+  state.cart.forEach((item) => {
+    const imageUrl = `assets/icons/${item.id}.svg`;
+    const altName = item.name;
```
This makes your code easy to understand further down the line

ChrisJS90 reviewed (commented):
Great work! Learned quite a bit of new functions here!

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+    // Create the paragraph tag and set its innerText.
+    const p = document.createElement("p");
+    // p.innerText = state.cart[index].item[index].name;
+    p.innerText = item.name;
```
Could you have used your altName variable here?

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+
+// Create the add to cart function.
+const addToCart = (item) => {
+  const copyStore = Object.assign({}, item);
```
Can you explain what this does?

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+};
+
+const priceUpdate = () => {
+  const total = state.cart.reduce(
+    (acc, item) => acc + item.price * item.quantity,
+    0
+  );
```
I also used reduce! so much easier than the methods I was thinking of before I found this

Rebsonau commented on index.js:
```diff
@@ -3,53 +3,185 @@ const state = {
+    0
+  );
+
+  const formatCurrency = new Intl.NumberFormat("en-GB", {
+    style: "currency",
+    currency: "GBP",
```
Very interesting this, I just used string interpolation and toFixed(2) to keep the decimal places

Rebsonau reviewed (commented):
Really good work and some interesting techniques in there that I'll need to look into

desokan committed:
Update README.md

desokan committed:
Update README.md

desokan committed:
Update README.md

In this project, I explored a common scenario in eCommerce, adding and removing items from the cart, and calculating the total. I used JS DOM, JavaScript, HTML, and CSS to build the project.
